src/renderers/WebGLRenderer.js

```diff
@@ -1339,6 +1339,12 @@ function WebGLRenderer( parameters ) {
 		let program = materialProperties.program;
 		let programChange = true;
 
+		// always update environment and fog - changing these trigger an initMaterial call, but it's possible that the program doesn't change
+
+		materialProperties.environment = material.isMeshStandardMaterial ? scene.environment : null;
+		materialProperties.fog = scene.fog;
```

Comment:
Shouldn't a different fog object detected by `getParameters()`?

Reply:
I think what this PR addresses is that even if it's a different object, the object type is the same?
But, honestly, I have not done a deep study on it.

Reply:
But if the fog type is the same and only the object differs, a shader recompilation is not necessary. Uniform updates should handle different fog values.

Reply:
@Oletus could you clarify?

Reply:
A live example that demonstrates the current issue would also help to clarify the change 😇 .

Reply:
I'm realizing now that I misunderstood the new comment in `WebGLRenderer`, sorry. 

The PR is intended to prevent additional calls of `initMaterial()` which are not necessary. These might happens since we compare objects in `setProgram()`. 

A live example would still help to fill the gaps in my mental model 😊 .

Reply:
Yes, your last comment is right. This patch makes sure materialProperties get updated if environment or fog is replaced with a different object of the same type. It could be possible to change webgl_materials_standard example to support changing to a different environment map, would you want me to do that?

Reply:
@Oletus That'd be great 👍